src/camera.ts

```diff
@@ -110,48 +121,58 @@ export default class Camera extends EventEmitter implements CameraState {
    * Method returning the coordinates of a point from the graph frame to the
    * viewport.
    *
-   * @param  {object} dimensions - Dimensions of the viewport.
-   * @param  {number} x          - The X coordinate.
-   * @param  {number} y          - The Y coordinate.
-   * @return {object}            - The point coordinates in the viewport.
+   * @param  {object} dimensions  - Dimensions of the viewport.
+   * @param  {object} coordinates - Coordinates of the point.
+   * @return {object}             - The point coordinates in the viewport.
    */
-
-  // TODO: assign to gain one object
-  // TODO: angles
-  graphToViewport(dimensions: { width: number; height: number }, x: number, y: number): Coordinates {
+  graphToViewport(dimensions: Dimensions, coordinates: Coordinates): Coordinates {
     const smallestDimension = Math.min(dimensions.width, dimensions.height);
 
-    const dx = smallestDimension / dimensions.width,
-      dy = smallestDimension / dimensions.height;
+    const dx = smallestDimension / dimensions.width;
+    const dy = smallestDimension / dimensions.height;
+    const ratio = this.ratio / smallestDimension;
+
+    // Align with center of the graph:
+    const x1 = (coordinates.x - this.x) / ratio;
+    const y1 = (this.y - coordinates.y) / ratio;
+
+    // Rotate:
+    const x2 = x1 * Math.cos(this.angle) - y1 * Math.sin(this.angle);
```

Comment:
Maybe it could be a good thing to cache `Math.cos(angle)` and `Math.sin(angle)` (as a variable in this function, or, if we are truly madmen, as a camera hidden prop) since this is a critical function? Could be overkill though...